app/src/main/java/gateway/controller/CtrlShareListWithWho.java

```diff
@@ -0,0 +1,83 @@
+package gateway.controller;
+
+import gateway.config.Config;
+import gateway.services.ServiceAuth;
+import gateway.services.ServiceAuth.ResUsername;
+import gateway.services.UtilValidator;
+import gateway.soap.request.ReqFile;
+import gateway.soap.response.ResShareListWithWho;
+import gateway.soap.response.ResStatus;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.util.UUID;
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+public class CtrlShareListWithWho
+{
+	public static ResShareListWithWho share_list_with_who (ReqFile args)
+	{
+		ResShareListWithWho resShareListWithWho = new ResShareListWithWho ();
```

Comment:
Rename `resShareListWithWho` to a shorter variable so it's easier to read. Something like 'res' or 's'.